Source/UserManagement/Domain/StaffUser/Commands/AddStaffUser.cs

```diff
@@ -6,23 +6,32 @@
 using System;
 using System.Collections.Generic;
 using Concepts;
+using doLittle.Commands;
 
-namespace Domain
+namespace Domain.StaffUser.Commands
 {
-    public class AddStaffUser
+    /// <summary>
+    /// Handles adding all kinds of staffusers.
+    /// It seems to that we have the option of this method, where fields can be empty.
+    /// Or have an add/remove command for each type of staffuser, which seems cumbersome and not dry.
```

Comment:
DRY never trumps your domain.  Anyway, DRY refers to logic, not data structures.  Don't be afraid to duplicate DTOs (which is what Commands and Events are).

If you want, you can create an abstract command and just inherit from it.  You can have a base validator that does all the work and can just be used by the concrete validators.

Reply:
I see, that's smart!